solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,421 @@
+    {
+        emit BurnIncentiveFeeUpdated(
+            _burnIncentiveFee,
+            newBurnIncentiveNetworkFee,
+            _maxBurnIncentiveFeeAmount,
+            newMaxBurnIncentiveFeeAmount
+        );
+
+        _burnIncentiveFee = newBurnIncentiveNetworkFee;
+        _maxBurnIncentiveFeeAmount = newMaxBurnIncentiveFeeAmount;
+    }
+
+    /**
+     * @dev returns the total amount of the burned governance tokens
+     *
+     * @return total amount of the burned governance tokens
+     */
+    function totalBurntAmount() external view returns (uint256) {
+        return _totalBurntAmount;
+    }
+
+    /**
+     * @dev converts the provided tokens to governance tokens and burns them
+     *
+     * @param tokens the tokens to convert
+     */
+    function burn(IERC20[] calldata tokens) external protected {
+        ITokenHolder feeWallet = networkFeeWallet();
+
+        // retrieve the burning strategy
+        Strategy memory strategy = burnStrategy(tokens, feeWallet);
+
+        // withdraw all token/ETH amounts to the contract
+        feeWallet.withdrawTokensMultiple(tokens, address(this), strategy.amounts);
+
+        // convert all amounts to the network token and record conversion amounts
+        IBancorNetwork network = bancorNetwork();
+
+        for (uint256 i = 0; i < strategy.paths.length; ++i) {
+            // avoid empty conversions
+            uint256 amount = strategy.amounts[i];
+            if (amount == 0) {
+                continue;
+            }
+
+            address[] memory path = strategy.paths[i];
+            IERC20 token = IERC20(path[0]);
+            uint256 value = 0;
+
+            if (token == _networkToken || token == _govToken) {
+                // if the source token is the network or the governance token, we won't try to convert it, but rather
+                // include its amount in either the total amount of tokens to convert or burn.
+                continue;
+            }
+
+            if (token != NATIVE_TOKEN_ADDRESS) {
+                // if the source token is a regular token, approve the converter to withdraw the token amount
+                ensureAllowance(token, network, amount);
+            } else {
+                // if the source token is actually an ETH reserve, make sure to pass its value to the network
+                value = amount;
+            }
+
+            // perform the actual conversion and optionally send ETH to the network
+            uint256 networkTokenConversionAmount =
+                network.convertByPath{ value: value }(path, amount, 1, address(this), address(0), 0);
+
+            emit Converted(token, amount, networkTokenConversionAmount);
+        }
+
+        // calculate the burn incentive fee and reduce it from the total amount to convert
+        NetNetworkTokenConversionAmounts memory netNetworkTokenConversionAmounts = netNetworkConversionAmounts();
+
+        // approve the governance token converter to withdraw the network token amount
```

Comment:
wrong comment (converter instead of network)

Reply:
👍

Reply:
still wrong :) "governance token network"

Reply:
![3a116884945f870924f1ffd3f36fc015](https://user-images.githubusercontent.com/2841614/112869254-9c4a3c80-90c5-11eb-82ac-4c7d34a662ea.png)